CMakeLists.txt

```diff
@@ -81,13 +81,19 @@ FOREACH(_build_type ${_d2_build_types})
 
     ADD_LIBRARY(${_lib} SHARED ${_files})
     FOREACH(_main ${_mains})
-	MESSAGE("-- Processing ${_main}")
-	GET_FILENAME_COMPONENT(_app_name ${_main} NAME_WE)
-    	SET(_exe "${_app_name}${_p}")
-    	MESSAGE("-- Configuring executable ${_exe}")
-    	ADD_EXECUTABLE(${_exe} ${_main})
-    	TARGET_LINK_LIBRARIES(${_exe} ${_lib})
-	D2K_SETUP_TARGET(${_exe} ${_BUILD_TYPE})
+      MESSAGE("-- Processing ${_main}")
+      GET_FILENAME_COMPONENT(_app_name ${_main} NAME_WE)
+      OPTION( ENABLE_EXEC_${_app_name} 
+              "Enable ${_app_name} executable" OFF)
```

Comment:
per chiarezza, la definizione delle opzioni sugli executables la metterei in un ciclo a parte